### ISSUE TITLE
feat: colNumber detect?

### ISSUE THREAD

**Issue by matteobertoldo** [FabiowQuixada/isml-linter#23]
Hi @FabiowQuixada! Thaankss for this nice `node` package 😊! 

I try to explain my request to create [Atom](https://atom.io) support for this tool.

I'm trying to implement `isml-linter` on Atom with [`language-sfcc`](https://github.com/matteobertoldo/language-sfcc/tree/feature/matteobertoldo%2Flinter) as well and it seems to work fine for now! There is a small problem currently. I am attaching the screenshot below:

<img width="1392" alt="hero-isml-linter" src="https://user-images.githubusercontent.com/15775323/109811626-5f586a80-7c2b-11eb-8ec6-fc44e1d733c3.png">

It would seem that the error output does not count the beginning of the column where the error starts ...

<img width="648" alt="col-rows" src="https://user-images.githubusercontent.com/15775323/109811645-667f7880-7c2b-11eb-89d1-4526a490ea7e.png">

in Output to console I have only the `lineNumber` which counts me where the errors are present in the rows ... but I don't find instead the `colNumber` which should check the beginning of the column for errors or any other type of diagnosis...how can I do?

Other thing, I couldn't change the indentation from `4` to `2` spaces ... and I don't understand how to configure the output as `warning` and not as `error`... are there any examples around?

Thanks so much.

**Comment by FabiowQuixada** [comment]
Hey @matteobertoldo , thanks for reaching out and I'm glad that Isml Linter can help you! :D

About the column number, indeed it is an important information to be present in the output, thanks for the suggestion! I'll work on it and will let you know.

About indentation, do you mean there is an issue with it? In the [Isml Linter npm page](https://www.npmjs.com/package/isml-linter), you can find the "indent" attribute, which should be at the root of the configuration object:

```
{
    indent: 2,
    rules: {
        ...
    }
} 
```

Please let me know if you find any issues with that.

About the occurrence level, you can find how to set it [here](https://github.com/FabiowQuixada/isml-linter/blob/master/docs/generic-rule-config.md), you will be able to iterate over the levels separately in the output's `error` , `warning` and `info` attributes.

Please let me know if you have further questions and thanks again! :D

**Comment by matteobertoldo** [comment]
@FabiowQuixada thank you so much! Let me know when the column count will be released so then I think this linter can be integrated into Atom as well.

As for the settings for parsing files, how does `isml-linter` intercept the various configurations? as parameters to parse the files I have only the file path and its content in the API ... but in Atom this seems to ignore when I change the values ...

I added the `.ismllinter.json` file in the root of my project...I added this settings...

```json
{
  "rootDir": "./",
  "ignore": ["spec", "hero.isml"],
  "rules": {
    "no-br": {},
    "enforce-require": {},
    "indent": {
      "value": 2
    }
  }
}
```

but it doesn't work...it always takes the default settings... 

<img width="458" alt="Schermata 2021-03-05 alle 17 05 24" src="https://user-images.githubusercontent.com/15775323/110141020-0298b380-7dd5-11eb-8874-36de3f463818.png">



it would be reasonable to integrate a third parameter in the [`parse`](https://github.com/FabiowQuixada/isml-linter/blob/9ffe4663d577db20108db17e3c771066f6192b39/src/publicApi.js#L10) API so that every time you correctly process settings other than the default ones? Or should I use the [`setConfig`](https://github.com/FabiowQuixada/isml-linter/blob/9ffe4663d577db20108db17e3c771066f6192b39/src/publicApi.js#L8) API?

**Comment by FabiowQuixada** [comment]
Hey @matteobertoldo , in fact, there was an issue with indent rule's "value" attribute, it should be now fixed as of version 5.30.4, release today. Please check and let me know if it works on your side as well.

You mentioned the configuration not being applied in the API, in previous versions it was needed to call `IsmlLinter.setConfig()` before running the linter, but I have implemented your suggestion and you can see in the [docs](https://github.com/FabiowQuixada/isml-linter/blob/master/docs/api.md) that the `parse` function now accepts a third, optional parameter `config`, thank you for pointing that out!

I have started the columnNumber feature, I think I'll be able to finish and release it next weekend, I will keep you posted on that.

Please let me know if you have any consideration to add or if any feature is not working properly and I'll be glad to check it.

Thank you!

**Comment by matteobertoldo** [comment]
Thank you very much @FabiowQuixada, really! I'll update you as soon as I update the implementation on Atom too!

Very kind!

**Comment by FabiowQuixada** [comment]
Hey @matteobertoldo , thanks for the feedback!

Release 5.35.0 adds the `columnNumber` attribute to the lint result, please check if it works as expected on your side. :)

The rule that might have some issue regarding the column number is the "eslint-to-isscript" rule, it currently reports the column number of the first character within the `<iscript>` tag, not the actual one of the error or warning. I still have to work with the integration with ESLint, but I'll let you know once it is done!

Thank you!

**Comment by FabiowQuixada** [comment]
Hey @matteobertoldo , the "eslint-to-isccript" issue I mentioned previously is now fixed, please check version 5.35.2 :D

Please let me know if it fits your needs :D

**Comment by matteobertoldo** [comment]
> Hey @matteobertoldo , the "eslint-to-isscript" issue I mentioned previously is now fixed, please check version 5.35.1 :D
> 
> Please let me know if it fits your needs :D

I have been very busy these past two weeks, next week I will update you with support for the latest fixes implemented. Thanks so much @FabiowQuixada

**Comment by FabiowQuixada** [comment]
Hey @matteobertoldo , don't worry, take your time, it has been a little rushy for me as well, so I understand. Take your time :D

**Comment by FabiowQuixada** [comment]
Hey @matteobertoldo , I will assume the "column number" feature is working as expected, based on the tests I performed during the feature development.

If you find any bug, please don't hesitate to open a new issue here!

Thanks!
